search-enterprise/latest/config/transactional.md

```diff
@@ -0,0 +1,382 @@
+
+AFTS and CMIS queries are parsed to an abstract form. This is then sent to an execution engine. There are two execution engines: the database and the Elasticsearch index. The default is to try the database first and fall back to the Solr index, if the query is not supported against the database. This is configurable for a search subsystem and per query using the Java API.
+
+To support TMDQ:
+
+* Alfresco Content Services supports TMDQ by default.
+
+## Features
+
+The following are the available feature of the transactional metadata query.
+
+* Transactional metadata query is supported for both Elasticsearch and noindex search subsystems.
+* Transactional metadata query does not support facets.
+* When you enable transactional metadata queries, a query is parsed to check if all of its parts are supported by the database-based query engine. If yes, the database is used automatically.
+* Using the database gives transactional consistency as opposed to the eventual consistency provided by Elasticsearch.
+* If you use the transactional metadata query with the noindex subsystem, the search functionality in Alfresco Share won't work as it relies on full text search.
+* Normally, a query will be executed against the database, if possible. Database execution of a query depends on the query itself. It also depends on the application of an optional patch to the database, which creates the required supporting database indexes. If the supporting indexes have been created, each index subsystem can be configured to:
+* perform transactional execution of queries;
+* execute queries transactionally, when possible, and fall back to eventual consistency; or
+* always execute eventual consistency.
+* When queries are executed against the database:
+* Hidden nodes will be returned by the database, as they are in Alfresco Content Services 5.0.
+* Large result sets are not supported because Alfresco Content Services does not evaluate permissions in query but as a post filter.
+* Counts will not reflect the number of nodes that match the query.
+* The `SearchParameters` and `QueryOptions` objects can be used to override this behaviour per query.
+Alfresco Content Services supports the execution of a subset of the CMIS Query Language (CMIS QL) and Alfresco Full Text Search (AFTS) queries directly against the database. Also, the noindex subsystem supports queries only against the database. This collection of features is called transactional metadata query (TMDQ).
+
+## Options supported by Query Languages
+
+Use this information to know what options are supported by the Public API, CMIS Query Language (QL), and Alfresco Full Text Search Query Language (FTS QL).
+
+### Public API and TMDQ
+
+From public API, anything that is not a simple query, a filter query, an option that affects these, or an option that affects what is returned for each node in the results, is not supported by TMDQ.
```

Comment:
Which public API? Is this talking about the v1 REST API?

Reply:
I have changed the references to be V1 Rest Api. What else could it be. Thanks.

Reply:
I changed all references from Public API to be V1 REST API.
No Public in the docs rechecked.